### PR TITLE
Update repositories.md

### PR DESCRIPTION
**Pull request by jonapich**
# Pull Request Check List

Resolves: some discussion in #3855 

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/repositories.md [hunk 1]
```diff
@@ -112,9 +112,27 @@ default = false
...
+[[tool.poetry.source]]
+name = "foo"
+url = "https://foo.bar/simple/"
+secondary = true
+```
+
 {{% warning %}}
 
-If package sources are defined for a project, these will take precedence over
+Reminder: If package sources are defined for a project, these will take precedence over
```
Comment (abn) [inline docs/repositories.md:135]: I think this is redundant over the existing warning box.

File: docs/repositories.md [hunk 2]
```diff
@@ -112,9 +112,27 @@ default = false
 secondary = false
 ```
 
+When a package source is defined like this, it takes precedence over [PyPI](https://pypi.org).
```
Comment (abn) [inline docs/repositories.md:115]: This is redundant I feel as this is said just below as well.

File: docs/repositories.md [hunk 3]
```diff
@@ -112,9 +112,27 @@ default = false
...
+```toml
+[tool.poetry.requirements]
+my-lib = { version = "*", source = "foo" }
+urllib3 = "*"
+
+[[tool.poetry.source]]
+name = "foo"
+url = "https://foo.bar/simple/"
+secondary = true
+```
```
Comment (abn) [inline docs/repositories.md:131]: I'd move this below the warning for continuity. And refer to the default section. Rephrase it something like -

```md
{{% note %}}
If you prefer to disable PyPI, you may choose to set one of your package sources to be the [default](#default-package-source).
{{% /note %}}
```

And move the exampe to the secondary package source section maybe?

**Review by abn — CHANGES_REQUESTED** [review]

**Comment by jonapich** [comment]
@abn I changed a few things. I added the `--secondary` to the first example too since that's what the wording was implying. 

However, this part in the secondary source section is basically false:

```
All package sources (including secondary sources) will be searched during the package lookup process. These network requests will occur for all sources, regardless of if the package is found at one or more sources.

If you wish to avoid this, you may explicitly specify which source to search in for a particular package.
```

You do not "avoid this" by adding the source. If you add the source to a package, the other dependencies _still use the secondary sources_. I would rephrase to:

```
You can force a dependency to use a specific package source (exclusively? or it will look it up first? not sure) by specifying which source to search in for a particular package.
```

**Review by abn — DISMISSED** [review]

File: docs/repositories.md [hunk 4]
```diff
@@ -112,23 +119,26 @@ default = false
...
 
-See [Supported Package Sources](#supported-package-sources) for source type specific information.
 
 {{% note %}}
 
-If your package source requires [credentials](#configuring-credentials) or
-[certificates](#certificates), please refer to the relevant sections below.
+If you prefer to disable [PyPI](https://pypi.org) completely, you may choose to set one of your package sources to be the [default](#default-package-source).
+
+To enable a package source only for a specific dependency, see [Secondary Package Sources](#secondary-package-sources).
```
[the remark this answers is not in the document]
Comment (jonapich) [reply docs/repositories.md]: I think this suggestion makes it unclear what this is about. I rephrased to:

> If you prefer to specify a package source for a specific dependency, see [Secondary Package Sources](#secondary-package-sources).

File: docs/repositories.md [hunk 5]
```diff
@@ -171,7 +181,18 @@ If you wish to avoid this, you may explicitly specify which source to search in
 package.
```
[the remark this answers is not in the document]
Comment (neersighted) [reply docs/repositories.md]: `source =` should restrict this to only that repository -- if we're doing otherwise, that's a bug.

File: docs/repositories.md [hunk 6]
```diff
@@ -112,23 +119,26 @@ default = false
...
 ```
 
-{{% warning %}}
-
-If package sources are defined for a project, these will take precedence over
-[PyPI](https://pypi.org). If you do not want this to be the case, you should declare **all** package
-sources to be [secondary](#secondary-package-sources).
-
-{{% /warning %}}
+Any package source defined like this takes precedence over [PyPI](https://pypi.org).
```
Comment (neersighted) [inline docs/repositories.md:122]:
```suggestion
Any package source not marked as `secondary` will take precedence over [PyPI](https://pypi.org).
```

File: docs/repositories.md [hunk 7]
```diff
@@ -167,11 +177,21 @@ All package sources (including secondary sources) will be searched during the pa
 process. These network requests will occur for all sources, regardless of if the package is
 found at one or more sources.
 
-If you wish to avoid this, you may explicitly specify which source to search in for a particular
-package.
+In order to limit the search for a specific package to a particular package source, you can explicitly specify what source to use.
```
Comment (neersighted) [inline docs/repositories.md:180]:
```suggestion
In order to limit the search for a specific package to a particular package repository, you can specify the source explicitly. This is strongly suggested for all private packages to avoid dependency confusion attacks.
```

**Review by neersighted — CHANGES_REQUESTED** [review]

File: docs/repositories.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jonapich) [reply docs/repositories.md]: It's difficult for me to assert this, because my private `pypi` is set to redirect to `pypi.org` on missing packages.

Since this closely touches security, there really should be a unit test that asserts it.

**Review by neersighted — APPROVED** [review]
